### ISSUE TITLE
Salt call cmdmod.exec_code_all fails on Windows minion due to invalid file mode

### ISSUE THREAD

**Issue by DarkKnightCZ** [saltstack/salt#34196]
### Description of Issue/Question
When executing any code via cmdmod.exec_code / exec_code_all on Windows minion, final file mode is 'w+tb' which is invalid on Windows OS. Execution results in exception
```python
 The minion function caused an exception: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1318, in _thread_return
        return_data = executor.execute()
      File "c:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 28, in execute
        return self.func(*self.args, **self.kwargs)
      File "c:\salt\bin\lib\site-packages\salt\modules\cmdmod.py", line 2296, in exec_code_all
        with salt.utils.fopen(codefile, 'w+t') as fp_:
      File "c:\salt\bin\lib\site-packages\salt\utils\__init__.py", line 1209, in fopen
        fhandle = open(*args, **kwargs)
    ValueError: Invalid mode ('w+tb')
```
### Setup
Any Windows minion.

### Steps to Reproduce Issue
Execute following command on Salt Master
``` salt 'windows-minion' cmd.exec_code_all powershell 'write-host "test string"' ```

### Versions Report
Master/Minion version: 2016.3.0
Also tested on 2015.8.10 with same result

**Comment by DarkKnightCZ** [comment]
Fixed in #34198. As cmdmod.exec_code_all contains following snippet
``` python
 # ensure 'binary' mode is always used on windows
    if kwargs.pop('binary', True):
        if is_windows():
            if len(args) > 1:
                args = list(args)
                if 'b' not in args[1]:
                    args[1] += 'b'
            elif kwargs.get('mode', None):
                if 'b' not in kwargs['mode']:
                    kwargs['mode'] += 'b'
            else:
                # the default is to read
                kwargs['mode'] = 'rb'
```
two easy fixes are possible.

1) Remove 't' mode from cmdmod.exec_code_all, which shouldn't be removed (after all it's text scripts what is executed) as it translates non-os newlines into proper ones
2) Remove 'b' mode, can be done by passing binary=False in kwargs

Second option is used in mentioned PR.

**Comment by Ch3LL** [comment]
@DarkKnightCZ thanks for the multiple PRs. We will go ahead nad get this closed up when the PRs are closed. Thanks!
